### ISSUE TITLE
salt '*' event.fire_master do not fire an event to all masters.

### ISSUE THREAD

**Issue by gashev** [saltstack/salt#40560]
Environment: two master servers: master-1 and master-2. Minions was installed on master-1 and master-2 servers.

Current minions configurations:
```
master:
  - "[fddc:3713:9ab3:0:0:17:0:50]:4506"
  - "[fddc:3713:9ab3:0:0:17:0:51]:4506"
ipv6: True
id: master-1
log_level: debug
log_level_logfile : debug
schedule:
 highstate:
  function: state.highstate
  minutes: 60
```
```
master:
  - "[fddc:3713:9ab3:0:0:17:0:51]:4506"
  - "[fddc:3713:9ab3:0:0:17:0:50]:4506"
ipv6: True
id: master-2
log_level: debug
log_level_logfile : debug
schedule:
 highstate:
  function: state.highstate
  minutes: 60
```

Salt version:
```
# salt --versions-report
           Salt: 2015.5.10
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
         Jinja2: 2.7.3
       M2Crypto: 0.20.2
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.0.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
        Tornado: Not Installed
        timelib: Not Installed
       dateutil: 1.4.1
```

I watch events on  master-1 and master-2 servers and exec on every master:
```
# salt-run state.event pretty=True
```

From the master-2 I fire an event and exec:
```
#  salt '*' event.fire_master '{"data":"my event data"}' 'tag'
master-1:
    True
master-2:
    True
```
on master-2 displayed events:
```
salt/event/new_client   {
    "_stamp": "2017-04-06T11:21:39.872605"
}
20170406142139901652    {
    "_stamp": "2017-04-06T11:21:39.902368",
    "minions": [
        "master-1",
        "master-2"
    ]
}
salt/job/20170406142139901652/new       {
    "_stamp": "2017-04-06T11:21:39.902946",
    "arg": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "fun": "event.fire_master",
    "jid": "20170406142139901652",
    "minions": [
        "master-1",
        "master-2"
    ],
    "tgt": "*",
    "tgt_type": "glob",
    "user": "root"
}
salt/job/20170406142139901652/ret/master-1     {
    "_stamp": "2017-04-06T11:21:41.065665",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-1",
    "jid": "20170406142139901652",
    "master_id": "master-2",
    "retcode": 0,
    "return": true,
    "success": true
}
salt/job/20170406142139901652/ret/master-2     {
    "_stamp": "2017-04-06T11:21:42.259498",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-2",
    "jid": "20170406142139901652",
    "master_id": "master-2",
    "retcode": 0,
    "return": true,
    "success": true
}
tag     {
    "_stamp": "2017-04-06T11:21:43.940430",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-1",
    "pretag": null,
    "tag": "tag"
}
tag     {
    "_stamp": "2017-04-06T11:21:45.134509",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-2",
    "pretag": null,
    "tag": "tag"
}
```
On master-1 displayed events:
```
tag     {
    "_stamp": "2017-04-06T11:21:43.935797",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-1",
    "pretag": null,
    "tag": "tag"
}
tag     {
    "_stamp": "2017-04-06T11:21:45.130143",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-2",
    "pretag": null,
    "tag": "tag"
}
```
I see events with tag on master-1 and master-2 servers.

Salt version:
```
# salt --versions-report
Salt Version:
           Salt: 2016.3.6

Dependency Versions:
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: 1.4.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, Aug  9 2016, 06:11:56)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.8 Santiago
        machine: x86_64
        release: 2.6.32-642.13.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.8 Santiago
```

From master-1 I fire an event; exec:
```
#  salt '*' event.fire_master '{"data":"my event data"}' 'tag'
master-2:
    True
master-1:
    True
```
Events on master-1:
```
20170406164623932566    {
    "_stamp": "2017-04-06T13:46:23.932905",
    "minions": [
        "master-1 ",
        "master-2 "
    ]
}
salt/job/20170406164623932566/new       {
    "_stamp": "2017-04-06T13:46:23.933716",
    "arg": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "fun": "event.fire_master",
    "jid": "20170406164623932566",
    "minions": [
        "master-1 ",
        "master-2 "
    ],
    "tgt": "*",
    "tgt_type": "glob",
    "user": "root"
}
salt/job/20170406164623932566/ret/master-2     {
    "_stamp": "2017-04-06T13:46:24.057939",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-2 ",
    "jid": "20170406164623932566",
    "master_id": "master-1",
    "retcode": 0,
    "return": true,
    "success": true
}
salt/job/20170406164623932566/ret/master-1   {
    "_stamp": "2017-04-06T13:46:24.059693",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-1 ",
    "jid": "20170406164623932566",
    "master_id": "master-1",
    "retcode": 0,
    "return": true,
    "success": true
}
tag     {
    "_stamp": "2017-04-06T13:46:24.060258",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-1 ",
    "pretag": null,
    "tag": "tag"
}
tag     {
    "_stamp": "2017-04-06T13:46:24.061481",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-2",
    "pretag": null,
    "tag": "tag"
}
```

No events on master-2 server. 

Events with tag received on master-1 server only.

The same if I fire event from the master-2 server. Events with tag received on master-1 server only.

From master-2 I fire an event; exec:
```
#  salt '*' event.fire_master '{"data":"my event data"}' 'tag'
master-2:
    True
master-1:
    True
```
Events on master-1 server:
```
tag     {
    "_stamp": "2017-04-06T13:53:31.047780",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-2 ",
    "pretag": null,
    "tag": "tag"
}
tag     {
    "_stamp": "2017-04-06T13:53:31.050061",
    "cmd": "_minion_event",
    "data": {
        "data": "my event data"
    },
    "id": "master-1 ",
    "pretag": null,
    "tag": "tag"
}
```
Events on master-2:
```
20170406165330885363    {
    "_stamp": "2017-04-06T13:53:30.886811",
    "minions": [
        "master-1 ",
        "master-2 "
    ]
}
salt/job/20170406165330885363/new       {
    "_stamp": "2017-04-06T13:53:30.887516",
    "arg": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "fun": "event.fire_master",
    "jid": "20170406165330885363",
    "minions": [
        "master-1 ",
        "master-2 "
    ],
    "tgt": "*",
    "tgt_type": "glob",
    "user": "root"
}
salt/job/20170406165330885363/ret/master-2     {
    "_stamp": "2017-04-06T13:53:31.044148",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-2 ",
    "jid": "20170406165330885363",
    "master_id": "master-2",
    "retcode": 0,
    "return": true,
    "success": true
}
salt/job/20170406165330885363/ret/master-1  {
    "_stamp": "2017-04-06T13:53:31.049697",
    "cmd": "_return",
    "fun": "event.fire_master",
    "fun_args": [
        {
            "data": "my event data"
        },
        "tag"
    ],
    "id": "master-1 ",
    "jid": "20170406165330885363",
    "master_id": "master-2",
    "retcode": 0,
    "return": true,
    "success": true
}
```
Events with tag received on master-1 server only.

So:
```
salt version 2015.5.10: salt '*' event.fire_master fire an event to all masters.
salt version 2016.3.6: salt '*' event.fire_master fire an event to one master only.
```
Is this a correct system behavior or error?

**Comment by Ch3LL** [comment]
I just tested this with 2016.11.3 and it is currently working. If you give that version a try does it work? But first can you confirm I understand the test case this correctly:

1. Run from master1: `salt '*' event.fire_master '{"data":"my event data"}' 'tag'`

2. Check events:
And on both master1 and master2 I see the following in the events (after running salt-run state.event pretty=True)

master1: `tag	{"_stamp": "2017-04-06T19:45:28.628811", "pretag": null, "cmd": "_minion_event", "tag": "tag", "data": {"data": "my event data"}, "id": "the_cake_is_a_lie"}`
master2: `tag     {"_stamp": "2017-04-06T19:46:10.218019", "pretag": null, "cmd": "_minion_event", "tag": "tag", "data": {"data": "my event data"}, "id": "the_cake_is_a_lie"}`

Where on 2016.3 it was broken so not both masters got that event?

**Comment by mattp-** [comment]
I fixed this in #49292
